protocol/decode.go

```diff
@@ -105,6 +105,21 @@ func (d *decoder) decodeArray(v value, elemType reflect.Type, decodeElem decodeF
 	}
 }
 
+func (d *decoder) decodeMap(v value, typ reflect.Type, decodeVal decodeFunc) {
+	if n := d.readInt32(); n < 0 {
+		v.setArray(array{})
+	} else {
+		m := array{val: reflect.MakeMap(typ)}
```

Comment:
Is this going to break if the `unsafe` build tag is specified (see `protocol/reflect_unsafe.go`)?